vavr/src/main/java/org/jdbi/v3/vavr/VavrTupleRowMapperFactory.java

```
@@ -53,38 +54,55 @@
+            if (anyColumnSet) {
+                Array<Optional<RowMapper<?>>> mappers = withConfiguredColumnName
+                        .map(t -> t._3.isDefined() ?
+                                getColumnMapperForDefinedColumn(t._1, t._3.get(), config) :
+                                getRowMapper(t._1, t._2, config));
+
+                boolean mappableWithConfigured = mappers.forAll(Optional::isPresent);
+                if (mappableWithConfigured) {
+                    return buildMapper(tupleClass, mappers);
+                }
+            }
 
-            final Array<Optional<RowMapper<?>>> colMappers = tupleTypes
-                    .map(t -> getColumnMapper(t._1, t._2 + 1, config));
+            Array<Optional<RowMapper<?>>> colMappers = tupleTypes
+                    .map(t -> getColumnMapper(t._1, t._2, config));
 
             boolean mappableByColumn = colMappers.forAll(Optional::isPresent);
             if (mappableByColumn) {
-                Array<? extends RowMapper<?>> cms = colMappers.map(Optional::get);
-                return Optional.of((rs, ctx) ->
-                        buildTuple(tupleClass, i -> cms.get(i).map(rs, ctx)));
+                return buildMapper(tupleClass, colMappers);
             }
 
-            final Array<Optional<RowMapper<?>>> rowMappers = tupleTypes
-                    .map(t -> getRowMapper(t._1, t._2 + 1, config));
+            Array<Optional<RowMapper<?>>> rowMappers = tupleTypes
+                    .map(t -> getRowMapper(t._1, t._2, config));
 
             boolean mappableByRowMappers = rowMappers.forAll(Optional::isPresent);
             if (mappableByRowMappers) {
-                Array<? extends RowMapper<?>> rms = rowMappers.map(Optional::get);
-                return Optional.of((rs, ctx) ->
-                        buildTuple(tupleClass, i -> rms.get(i).map(rs, ctx)));
+                return buildMapper(tupleClass, rowMappers);
             }
```

Similar to above, let's throw an informative exception here and try to give users a hint what they did wrong.

For each tuple element with no row mapper:
- If there is a column mapper for that type: "Missing column name for tuple element _2 of type Foo"
- Otherwise: "Missing row mapper for tuple element _2 of type Foo"

My reasoning for these exceptions is: if the mapped type is some tuple, then this class should either return a tuple mapper or throw an exception so the users know what they did wrong. We only return `Optional.empty()` for non-tuples.